### PR TITLE
Shereen_Created Routing and controller function for Add Tool

### PR DESCRIPTION
**Pull request by pshereen**
Created Routing and controller function for Add Tool request.

# Description
Created Routing and controller function for Add Tool request.
Implements # (WBS) : 4.5.8

## Related PRS (if any):
This frontend PR is related to the #XXX backend PR.
To test this backend PR you need to checkout the[ #2306](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2306) frontend PR.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Follow the steps in frontend PR https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2306

### CODE REVIEW

**Review by Kavil-Jain-514 — CHANGES_REQUESTED** [review]
A few changes need to be made. 
Added comments and images on the front-end PR https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2306#pullrequestreview-2091439174

**Review by Sandhya1236 — APPROVED** [review]
I have successfully tested the updated Add Tool component, which now includes an API call to pass form data to the backend, form validation, and updates to store the data in the Redux store. This frontend PR, related to backend PR #970, was tested by checking out the current branch, running npm install, and starting the backend from the dev branch. Navigating to http://localhost:3000/BMdashboard/tools/add and logging in if prompted, I entered data into the form and verified that clicking the submit button saved all form data successfully. Additionally, if the tool name already existed in the database, the appropriate error message, 'Oops!! Tool already exists!', was displayed, confirming that all functionalities worked as expected.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/79351059/7fa5d23d-9522-4f15-9040-3d20ed155f03

**Review by jinxyou — CHANGES_REQUESTED** [review]
I left a review at https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2306#pullrequestreview-2120947694

File: src/models/bmdashboard/buildingInventoryType.js
Comment (Parth-tech) [inline src/models/bmdashboard/buildingInventoryType.js:78]: Hi Shereen,
Can we get rid of comments on line 77 & 78 if they do not serve any useful purpose?

File: src/controllers/bmdashboard/bmInventoryTypeController.js
Comment (Parth-tech) [inline src/controllers/bmdashboard/bmInventoryTypeController.js:198]: I believe we do not need try-catch block as all the code that is present within the try catch block is using `.then() and .catch()`.

**Review by Parth-tech — COMMENTED** [review]
Hey Shereen,
The backend looks good to me. But I have requested changes of the front-end so I will approve the backend once the front-end is ready to be approved.

[Frontend Review](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2306#pullrequestreview-2149084439)

**Review by DiyaWadhwani — APPROVED** [review]
Hi Shereen,

I have dropped a comment on your frontend PR and requested a few UI changes.
Kindly review them
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2306#pullrequestreview-2149589146
Thank you.

**Review by HowieMiao — APPROVED** [review]
Changes in place are functional, only numbers are accepted for phone number and rental date validation looks good. Cannot add duplicate tools

image

**Review by jinxyou — APPROVED** [review]
I approved the PR in frontend at https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2306#pullrequestreview-2161378244

**Review by nikhilpittala16 — APPROVED** [review]
I have tested the PR and left a review on the front-end. link can be found here:https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2306#pullrequestreview-2161387339

**Review by Parth-tech — APPROVED** [review]
Hey Sheeren,
Great job done. I have left a review on the front-end, [review link.](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2306#pullrequestreview-2161469343)

**Review by Kavil-Jain-514 — DISMISSED** [review]
All the fields are working fine, but there is one change. 
When the user selects Rental for the Purchase/Rental field and again changes the dropdown to purchase, then while submitting the form, the Return date says required as technically it should not show that error message.

Please take a look at the video below for reference.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112303944/2529cd49-aaaf-4c48-9c06-6acb5a605b36

**Comment by pshereen** [comment]
> A few changes need to be made. Added comments and images on the front-end PR [OneCommunityGlobal/HighestGoodNetworkApp#2306 (review)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2306#pullrequestreview-2091439174)

Addressed

**Comment by pshereen** [comment]
> I left a review at [OneCommunityGlobal/HighestGoodNetworkApp#2306 (review)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2306#pullrequestreview-2120947694)

Addressed

**Comment by pshereen** [comment]
> A few changes need to be made. Added comments and images on the front-end PR [OneCommunityGlobal/HighestGoodNetworkApp#2306 (review)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2306#pullrequestreview-2091439174)

Addressed

**Comment by pshereen** [comment]
> All the fields are working fine, but there is one change. When the user selects Rental for the Purchase/Rental field and again changes the dropdown to purchase, then while submitting the form, the Return date says required as technically it should not show that error message.
> 
> Please take a look at the video below for reference.
> 
>  PR.2306-970.mp4

Addressed

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**
